src/Providers/Anthropic/Handlers/Text.php

```diff
@@ -4,43 +4,56 @@
 
 namespace EchoLabs\Prism\Providers\Anthropic\Handlers;
 
-use EchoLabs\Prism\Exceptions\PrismException;
+use EchoLabs\Prism\Contracts\PrismRequest;
 use EchoLabs\Prism\Providers\Anthropic\Maps\FinishReasonMap;
 use EchoLabs\Prism\Providers\Anthropic\Maps\MessageMap;
 use EchoLabs\Prism\Providers\Anthropic\Maps\ToolChoiceMap;
 use EchoLabs\Prism\Providers\Anthropic\Maps\ToolMap;
-use EchoLabs\Prism\Text\Request;
+use EchoLabs\Prism\Text\Request as TextRequest;
 use EchoLabs\Prism\ValueObjects\ProviderResponse;
 use EchoLabs\Prism\ValueObjects\ResponseMeta;
 use EchoLabs\Prism\ValueObjects\ToolCall;
 use EchoLabs\Prism\ValueObjects\Usage;
-use Illuminate\Http\Client\PendingRequest;
-use Illuminate\Http\Client\Response;
-use Throwable;
 
-class Text
+/**
+ * @template TRequest of TextRequest
+ */
+class Text extends AnthropicHandlerAbstract
 {
-    public function __construct(protected PendingRequest $client) {}
-
-    public function handle(Request $request): ProviderResponse
+    /**
+     * @param  TextRequest  $request
+     * @return array<string, mixed>
+     */
+    #[\Override]
+    public static function buildHttpRequestPayload(PrismRequest $request): array
     {
-        try {
-            $response = $this->sendRequest($request);
-            $data = $response->json();
-
-        } catch (Throwable $e) {
-            throw PrismException::providerRequestError($request->model, $e);
+        if (! $request instanceof TextRequest) {
+            throw new \InvalidArgumentException('Request must be an instance of '.TextRequest::class);
         }
```

Comment:
Could we maybe abstract this to a trait? `$request->is(Structured::class)` or something like that?